src/js/stores/MarathonStore.js

```diff
@@ -160,6 +160,12 @@ var MarathonStore = Store.createStore({
   },
 
   processMarathonGroups: function (data) {
+    data.filterProperties = {
+      name: function (item) {
+        return item.id;
+      }
+    };
+
```

Comment:
I've just refactored the List and Tree struct (https://github.com/dcos/dcos-ui/pull/75), as it has always bothered me, that you have to pass the filter properties as constructor parameter.  With the change in place you're able to pass a definition of filterProperties whenever you filter items by text.

BTW: If you don't want to use the fancy `filterItemsByText` method, you could use the plain `filterItems` method. `filterItems` works like the `Array.prototype.filter` method but is applied to the whole tree. This would allow you, for exmaple, to use something like `fuzzaldrin` to implement a fuzzy filter/search in a few lines.